spec/services/end_of_cycle_timetable_spec.rb

```diff
@@ -140,4 +141,84 @@ def create_application_for(recruitment_cycle_year)
       end
     end
   end
+
+  context 'application form is in the apply1 state' do
+    let(:application_form) { build_stubbed(:application_form) }
+
+    context 'when the date is after the apply1 submission deadline' do
+      it 'returns false' do
+        Timecop.travel(EndOfCycleTimetable.apply_1_deadline + 1.day) do
+          expect(execute_service).to eq false
+        end
+      end
+    end
+
+    context 'when the date is before the apply1 submission deadline' do
+      it 'returns true' do
+        Timecop.travel(EndOfCycleTimetable.apply_1_deadline) do
+          expect(execute_service).to eq true
+        end
+      end
+    end
+
+    context 'when the date is post find reopening' do
+      it 'returns true' do
+        Timecop.travel(EndOfCycleTimetable.find_reopens) do
+          expect(execute_service).to eq true
+        end
+      end
+    end
+  end
+
+  context 'application form is in the apply again state' do
+    let(:application_form) { build_stubbed(:application_form, phase: :apply_2) }
+
+    context 'when the date is after the apply again submission deadline' do
+      it 'returns false' do
+        Timecop.travel(EndOfCycleTimetable.apply_2_deadline + 1.day) do
+          expect(execute_service).to eq false
+        end
+      end
+    end
+
+    context 'when the date is before the apply again submission deadline' do
+      it 'returns true' do
+        Timecop.travel(EndOfCycleTimetable.apply_2_deadline) do
+          expect(execute_service).to eq true
+        end
+      end
+    end
+
+    context 'when the date is post find reopening' do
+      it 'returns true' do
+        Timecop.travel(EndOfCycleTimetable.apply_reopens) do
+          expect(execute_service).to eq true
+        end
+      end
+    end
+  end
+
+  context 'application form is from a previous recruitment cycle' do
+    let(:application_form) { build_stubbed(:application_form, recruitment_cycle_year: 2020) }
+
+    it 'returns false' do
+      Timecop.travel('2021-02-03') do
+        expect(execute_service).to eq false
+      end
+    end
+  end
```

Comment:
Perhaps all of this should be in a `describe 'can_add_course_choice?'` block, so that:
- the testing style is consistent
- the `let(:execute_service)` can be moved locally to where it's needed